mtls_server/utils.py

```diff
@@ -185,3 +195,33 @@ def import_and_trust(key_data, gpg):
 def create_dir_if_missing(path):
     if not os.path.isdir(path):
         os.makedirs(path)
+
+def time_in_range(start, end, t):
+    """Return true if t is in the range [start,end]"""
+    if start <= end:
+        return start <= t <= end
+    else:
+        return start <= t or t <= end
+
+def has_user(gpg, fingerprint):
+    keys = gpg.list_keys(keys=fingerprint)
+    if len(keys) == 0:
+        return False
+    return True
+
+def add_and_trust_user(gpg, fingerprint, keyserver="keyserver.ubuntu.com"):
+    logger.info(f"Retrieving key {fingerprint} from {keyserver}")
+    result = gpg.recv_keys(
+        keyserver,
+        fingerprint,
+    )
```

Comment:
Does this imply that if you're not on the keyserver you can't be added?

Reply:
Currently yes

Reply:
This has been a known issue, you either need to add them to the keyring manually, or include them in the seed folder.

We'll need to make an issue to allow for uploading keys are part of the post request.